src/Modules/Data/Commands.Data/AddPowerBIDataset.cs

```diff
@@ -0,0 +1,86 @@
+﻿/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Text;
+using Microsoft.PowerBI.Common.Abstractions;
+using Microsoft.PowerBI.Common.Abstractions.Interfaces;
+using Microsoft.PowerBI.Common.Api.Datasets;
+using Microsoft.PowerBI.Common.Api.Workspaces;
+using Microsoft.PowerBI.Common.Client;
+
+namespace Microsoft.PowerBI.Commands.Data
+{
+    [Cmdlet(CmdletVerb, CmdletName, DefaultParameterSetName = DatasetParameterSetName)]
+    [OutputType(typeof(Dataset))]
+    public class AddPowerBIDataset : PowerBIClientCmdlet, IUserScope
+    {
+        public const string CmdletVerb = VerbsCommon.Add;
+        public const string CmdletName = "PowerBIDataset";
+
+        #region ParameterSets
+        private const string DatasetParameterSetName = "Dataset";
+        private const string WorkspaceParameterSetName = "Workspace";
+        #endregion
+
+        #region Parameters
+        [Parameter(Mandatory = true)]
+        public Dataset Dataset { get; set; }
+       
+        [Alias("GroupId")]
+        [Parameter(Mandatory = false, ParameterSetName = DatasetParameterSetName)]
+        public Guid WorkspaceId { get; set; }
+
+        [Alias("Group")]
+        [Parameter(Mandatory = true, ParameterSetName = WorkspaceParameterSetName, ValueFromPipeline = true)]
+        public Workspace Workspace { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public PowerBIUserScope Scope { get; set; } = PowerBIUserScope.Individual;
```

Comment:
We could remove the Scope for now as it not used, we can add back once we support an Admin API

Reply:
Okay, but we had discussion before to put it in even though we dont support it yet. Please let me know which we really want. I don't have any issue removing it though.

Reply:
For cmdlets that create things we may never have an Admin API for. The question will be for this, does it benefit the admin could create data and create a dataset and put that data anywhere without them separately adding permission to do this operation (2-step approach). We can always add Scope back in if we create the API.